pkg/api-server/inspect_endpoints.go

```diff
@@ -52,23 +52,26 @@ func (s *simpleMatchedPolicyGetter) Get(ctx context.Context, dataplaneKey core_m
 		return nil, err
 	}
 
-	mesh := core_mesh.NewMeshResource()
-	if err := s.resManager.Get(ctx, mesh, store.GetByKey(dataplaneKey.Mesh, core_model.NoMesh)); err != nil {
-		return nil, err
-	}
+	meshContextBuilder := xds_context.NewMeshContextBuilder(
+		s.resManager,
+		server.MeshResourceTypes(server.HashMeshExcludedResources),
+		net.LookupIP,
+		s.cfg.Multizone.Zone.Name,
+		vips.NewPersistence(s.resManager, s.cfgManager),
+		s.cfg.DNSServer.Domain,
+	)
 
-	dataplanes, err := xds_topology.GetDataplanes(core.Log, ctx, s.resManager, net.LookupIP, dataplaneKey.Mesh)
+	meshContext, err := meshContextBuilder.Build(ctx, dataplaneKey.Mesh)
 	if err != nil {
 		return nil, err
 	}
 
 	// todo(lobkovilya): share DataplaneProxyBuilder with xDS code (instead of creating a new one)
-	proxyBuilder := sync.DefaultDataplaneProxyBuilder(s.resManager, s.resManager, net.LookupIP,
-		datasource.NewDataSourceLoader(s.resManager), *s.cfg, s.cfgManager, &fakeMetadataTracker{}, envoy.APIV3)
-	proxy, err := proxyBuilder.Build(core_model.MetaToResourceKey(dataplane.GetMeta()), &xds_context.Context{Mesh: xds_context.MeshContext{
-		Resource:   mesh,
-		Dataplanes: dataplanes,
-	}})
+	proxyBuilder := sync.DefaultDataplaneProxyBuilder(datasource.NewDataSourceLoader(s.resManager),
+		*s.cfg, &fakeMetadataTracker{}, envoy.APIV3)
+	proxy, err := proxyBuilder.Build(core_model.MetaToResourceKey(dataplane.GetMeta()), &xds_context.Context{
```

Comment:
I think you can modify DataplaneProxyBuilder signature to take MeshContext as a second argument

Reply:
Will be fixed here https://github.com/kumahq/kuma/pull/3568/files#r786078108